lib/common/threading.h

```diff
@@ -37,7 +37,15 @@ extern "C" {
 #  define WIN32_LEAN_AND_MEAN
 #endif
 
+#ifdef ERROR
+#  undef ERROR   /* reported already defined on VS 2015 (Rich Geldreich) */
+#endif
 #include <windows.h>
+#ifdef ERROR
+#  undef ERROR   /* reported already defined on VS 2015 (Rich Geldreich) */
+#endif
+#define ERROR(name) ZSTD_ERROR(name)
```

Comment:
In which case is the `ERROR()` defined here useful ?

Reply:
Yeah, if `zstd_internal.h` is included after, then it isn't useful. But if it is included before, we've just undefined `ERROR`, so we need to redefine it.